src/java/org/apache/cassandra/db/SerializationHeader.java

```diff
@@ -384,6 +386,97 @@ public EncodingStats getEncodingStats()
         }
     }
 
+    public interface ParameterizedSerializer<P>
```

Comment:
can we get serde tests for these new serializers?  been adding them to other serializers in https://github.com/apache/cassandra/pull/3954/ and keep finding bugs, so would be good to get the tests in with the new code